File: job-board-net/src/JobBoard/Models/Job.cs
namespace JobBoard.Models;

public class Job
{
    public int Id { get; set; }
    public string Title { get; set; } = "";
    public string Company { get; set; } = "";
    public string Location { get; set; } = "";
    public string Description { get; set; } = "";
    public JobType EmploymentType { get; set; }
    public RemoteType LocationType { get; set; }
    public string ApplicationUrl { get; set; } = "";
    public DateTime PostedDate { get; set; } = DateTime.UtcNow;
}

public enum JobType { FullTime, PartTime, Contract }
public enum RemoteType { Remote, Onsite, Hybrid }

File: job-board-react/backend/src/validation/job.ts
import { z } from "zod";

export const employmentTypes = ["FullTime", "PartTime", "Contract"] as const;
export const locationTypes = ["Remote", "Onsite", "Hybrid"] as const;

export const createJobSchema = z.object({
  title: z.string().min(1),
  company: z.string().min(1),
  location: z.string().min(1),
  description: z.string().min(1),
  employmentType: z.enum(employmentTypes),
  locationType: z.enum(locationTypes),
  applicationUrl: z.string().url(),
});

export type CreateJobInput = z.infer<typeof createJobSchema>;
